Движение в сторону полноценной библиотеки

lmasikl opened the pull request.

lmasikl committed:
Добавлены пропущенные зависимости

lmasikl committed:
Выделены общие части для двух скриптов

lmasikl committed:
Расширен .gitignore

empenoso commented:
Пока до выхода статьи в этот четверг https://habr.com/ru/articles/883704/ ничего менять не буду

lmasikl committed:
Merge branch 'empenoso:main' into proposal

lmasikl committed:
Скрипт поиска по критериям выделен в расширяемое приложение

- Скрипт обернут классом приложения для дальнейшей декомпозиции
- Запись данных полученных с помощью скрипта вынесена в плагин
  для добавления в будущем возможности вывода в альтернативные форматы
- Добавлены основвные типы данных для более простого чтения кода и
  минимизации ошибок при передаче данных между функциями
- Добавлено простейшее меню запуска приложения для того чтобы обеспечить
  единую точку входа в приложение

empenoso commented:
Скачал весь измененный код и VSCode что-то ругается.

Вот это место: https://github.com/lmasikl/moex-bond-search-and-analysis/blob/proposal/1_bonds_search%20by%20criteria.py#L21

![изображение](https://github.com/user-attachments/assets/4e38fc66-9389-446f-9322-550bed3b36cb)

lmasikl commented:
> Скачал весь измененный код и VSCode что-то ругается.
> 
> Вот это место: https://github.com/lmasikl/moex-bond-search-and-analysis/blob/proposal/1_bonds_search%20by%20criteria.py#L21

Потому что в системных путях нету папки `/path/to/src`. Если их добавить руками, то не будет ругаться. Код работает, потому что при старте скрипта данный путь добавляется до импортов. Другой вариант: создавать виртуальное окружение как описано в альтернативной установке.

empenoso commented:
Оказывает еще надо `pip install humanize` чтобы заработало
